pkg/datapath/iptables/iptables.go

```diff
@@ -694,6 +711,70 @@ func (m *IptablesManager) iptProxyRules(cmd string, proxyPort uint16, ingress bo
 	return nil
 }
 
+func noTrackRules(prog string, cmd string, IP string, port *lb.L4Addr, ingress bool) error {
+	protocol := strings.ToLower(port.Protocol)
+	p := strconv.FormatUint(uint64(port.Port), 10)
+	if ingress {
+		if _, err := runProgCombinedOutput(prog, []string{"-t", "raw", cmd, "PREROUTING", "-p", protocol, "-d", IP, "--dport", p, "-j", "NOTRACK"}, false); err != nil {
+			return err
+		}
+		if _, err := runProgCombinedOutput(prog, []string{"-t", "filter", cmd, "INPUT", "-p", protocol, "-d", IP, "--dport", p, "-j", "ACCEPT"}, false); err != nil {
+			return err
+		}
+		if _, err := runProgCombinedOutput(prog, []string{"-t", "raw", cmd, "OUTPUT", "-p", protocol, "-d", IP, "--dport", p, "-j", "NOTRACK"}, false); err != nil {
+			return err
+		}
+	} else {
+		if _, err := runProgCombinedOutput(prog, []string{"-t", "raw", cmd, "OUTPUT", "-p", protocol, "-s", IP, "--sport", p, "-j", "NOTRACK"}, false); err != nil {
+			return err
+		}
+		if _, err := runProgCombinedOutput(prog, []string{"-t", "filter", cmd, "OUTPUT", "-p", protocol, "-s", IP, "--sport", p, "-j", "ACCEPT"}, false); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func InstallNoTrackRules(IP string, port string, ipv6 bool) error {
+	prog := "iptables"
+	if ipv6 {
+		prog = "ip6tables"
+	}
+	ports, err := dnsPorts(port)
+	if err != nil {
+		return err
+	}
+	for _, p := range ports {
+		if err := noTrackRules(prog, "-A", IP, p, true); err != nil {
+			return err
+		}
+		if err := noTrackRules(prog, "-A", IP, p, false); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func RemoveNoTrackRules(IP string, port string, ipv6 bool) error {
+	prog := "iptables"
+	if ipv6 {
+		prog = "ip6tables"
+	}
+	ports, err := dnsPorts(port)
+	if err != nil {
+		return err
+	}
```

Comment:
This seems late for input validation, see [0]

Reply:
Maybe we should keep these validation in case other use cases don't validate before reaching datapath?

Reply:
In general we aim not to have additional layers of code that try to cover for other layers not doing their job right; if the outer layers are not handling the inputs appropriately, then we should fix those outer layers. That will keep the core code here simpler.

I believe there should be other ways to ensure that the input is validated before it hits this layer, for instance if the port was a dedicated type (ie not generic unvalidated string) at this point then the caller would be forced to check and convert to the dedicated type before passing nonsense into this function.

Reply:
Changed input type to uint16, no longer needs validate here.